vignettes/Pseudobulking-and-Filtered-Contrasts.Rmd

```diff
@@ -0,0 +1,386 @@
+
+## Why should we use Pseudobulking? 
+
+In scRNASeq analyses, it is very common to use unsupervised clustering approaches to create clusters of cells for which their gene expression is grossly different. These cell groupings sometimes correspond to distinct cellular phenotypes, but other times clusters capture transient cell states such as division or metabolic shifts. Once these clusters are obtained, one generally runs a variation of Seurat's `FindMarkers()` to perform a wilcoxon test between two clusters, or a one-versus-all test to compare a single cluster to the rest of the data grouped together. This approach works well for determining the top 10/50/100 genes that define the population, but falls short when one asks "What are the differentially expressed genes?". 
+
+
+Differential gene expression usually considers the "strength" of expression $Log_2(\dfrac{Expression_{experimental}}{Expression_{control}})$ and the p value as computed via a hypothesis test versus a null distribution. Frequently, you will discover that the p values obtained via the clustering and `FindMarkers()` method will be infinitesimally small. This is due to a phenomenon known as double dipping, wherein the clustering was determined through differences in gene expression, and then the genes were tested for "how different is the gene's expression between clusters" using a null hypothesis that is no longer feasible (e.g. using $Expression_{cluster_1} = Expression_{cluster_2}$ as a null hypothesis when cluster membership maximizes $Expression_{cluster_1} \neq Expression_{cluster_2}$). 
+
+
+Pseudobulking offers a way to statistically determine gene sets that are generally congruent with one's expectations of type I and type II error control. 
+
+For more on double dipping and approaches to circumvent it, please read https://arxiv.org/abs/2301.07276 and related works from the Witten lab.
+
+## How to Pseudobulk
+
+```{r start pseudobulking, fig.width = 10}
+#Load seurat object
+seuratObj <- suppressMessages(suppressWarnings(
+    Seurat::UpdateSeuratObject(readRDS("../tests/testdata/seuratOutput.rds"))))
+
+#assign some random experimental variables: vaccine groups, experimental time points, and subject Ids. 
+seuratObj@meta.data[,"vaccine_cohort"] <- base::rep(c("control", "vaccineOne", "vaccineTwo", "unvax"), length.out = length(colnames(seuratObj)))
+  seuratObj@meta.data[,"timepoint"] <- base::rep(c("baseline", "necropsy", "day4"), length.out = length(colnames(seuratObj)))
+  seuratObj@meta.data[,"subject"] <- base::sample(c(1,2,3,4), size = 1557, replace = T)
+  
+  seuratObj@meta.data$sample_identifier <- paste0("Subject-", seuratObj$subject, '-', seuratObj$vaccine_cohort, "-", seuratObj$timepoint)
+
+#set up pseudobulking
+pbulk <- suppressWarnings(PseudobulkSeurat(seuratObj, groupFields = c("vaccine_cohort", "timepoint", "sample_identifier")))
```

Comment:
I'm not seeing anything wrong here - this should have a 30k-ish row, 48ish column count matrix.